lib/android-tools-info.ts

```diff
@@ -0,0 +1,209 @@
+
+				let infoData: IAndroidToolsInfoData = Object.create(null);
+				infoData.androidHomeEnvVar = this.getAndroidHome();
+				infoData.compileSdkVersion = this.getCompileSdk().wait();
+				infoData.buildToolsVersion = this.getBuildToolsVersion().wait();
+				infoData.targetSdkVersion = this.getTargetSdk().wait();
+				infoData.supportLibraryVersion = this.getAndroidSupportLibVersion().wait();
+
+				this.toolsInfo = infoData;
+			}
+
+			return this.toolsInfo;
+		}).future<IAndroidToolsInfoData>()();
+	}
+
+	public validateInfo(options?: {showWarningsAsErrors: boolean, validateTargetSdk: boolean}): IFuture<void> {
+		return (() => {
+			let toolsInfoData = this.getToolsInfo({showWarningsAsErrors: options && options.showWarningsAsErrors}).wait();
+			if(!toolsInfoData.androidHomeEnvVar || !this.$fs.exists(toolsInfoData.androidHomeEnvVar).wait()) {
+				this.printMessage("ANDROID_HOME environment variable is not set or it points to not existing directory. You cannot use gradle without setting it.",
+					"Make sure you have set ANDROID_HOME and it points the root of your Android SDK installation directory.");
+			}
+
+			if(!toolsInfoData.compileSdkVersion) {
+				this.printMessage(`Could not find supported Android target for compilation. Please install at least: ${AndroidToolsInfo.MIN_REQUIRED_COMPILE_TARGET}.` +
+							" Make sure you have the latest Android tools installed as well.",
+							'Run "android" from your command-line to install required Android SDK version.');
+			}
+
+			if(!toolsInfoData.buildToolsVersion) {
+				this.printMessage(`You have to install Android Build Tools and the version should be in the following range: '${this.getBuildToolsRange()}'.`,
+				'Run "android" from your command-line to install required Android Build Tools.');
+			}
+
+			if(!toolsInfoData.supportLibraryVersion) {
+				this.printMessage(`You have to install Android Support Library in the following range ${this.getAppCompatRange().wait() || ">=" + AndroidToolsInfo.MIN_REQUIRED_COMPILE_TARGET}}.`,
+					'Run "android" from your command-line to install Android Support Library.');
+			}
+
+			if(options && options.validateTargetSdk) {
+				let targetSdk = toolsInfoData.targetSdkVersion;
+				let newTarget = `${AndroidToolsInfo.ANDROID_TARGET_PREFIX}-${targetSdk}`;
+				if(!_.contains(AndroidToolsInfo.SUPPORTED_TARGETS, newTarget)) {
+					let supportedVersions = AndroidToolsInfo.SUPPORTED_TARGETS.sort();
+					let minSupportedVersion = this.parseAndroidSdkString(_.first(supportedVersions));
+
+					if(targetSdk && (targetSdk < minSupportedVersion)) {
+						this.printMessage(`The selected target SDK ${newTarget} is not supported. You should target at least ${minSupportedVersion}.`);
+					} else if(!targetSdk || targetSdk > this.getMaxSupportedVersion()) {
+						this.$logger.warn(`The selected Android target '${newTarget}' is not verified as supported. Some functionality may not work as expected.`);
```

Comment:
Support for the selected Android target SDK ${newTarget} is not verified. Your Android app might not work as expected.